README.md

```diff
@@ -1 +1,118 @@
-# angular-using-compile-link-readme
\ No newline at end of file
+# Using the compile/link functions
+
+## Overview
+
+We used the `link` function in our previous lab, however we don't actually know much about it. Let's look into `link` and the power it provides us.
+
+## Objectives
+
+- Describe the link function
+- Describe the compile function
+- Describe the Directive's lifecycle
+
+## Directive Lifecycle
+
+Each directive has a lifecycle. Up until now we've only been using one part of the lifecycle - the controller. The controller is initiated and ran just before the directive is about to be mounted into the DOM. This is so we can manipulate any values that we need in time for the rendering.
+
+However, it might come to the point where we need to actually do raw DOM manipulation on the directive itself. One problem - the controller is initiated *before* the directive is in the DOM. What do we do?
+
+
+### `link`
+
+This is where the `link` function comes in. If we specify this, we get a function executed *after* the directive has been mounted in the DOM. We also then get the DOM element passed through to us as the second parameter.
+
+```js
+function someDirective() {
+  return {
+    template: [
+	  '<div>',
+	    'Hello!',
+	  '</div>'
+	].join(''),
+    link: function ($scope, $element, $attrs) {
+
+    }
+  };
+}
+
+angular
+  .module('app', [])
+  .directive('someDirective', someDirective);
+```
+
+You can see above that we get three parameters (they must be in that order!). `$scope` - much like what we can inject into our controllers. `$element`, our DOM mounted element, and `$attrs`, an object of all the attributes used when the directive is initiated.
+
+### `compile`
+
+Compile allows us to manipulate the element before it is inserted into the DOM. This runs in between the `controller` and the `link` function. This gives us access to the unmounted DOM node, and the attributes.
```

Comment:
Why would you want to do this? What's an example of when you'd use a compile function?